### DISCUSSION TITLE
TabPanel transition

### DISCUSSION THREAD

**Discussion by wan54** [chakra-ui/chakra-ui#3400] [Q&A]
Hi there,

Is there an example for transitioning between tab panels? I'm looking for a way to do "fade out" on the old panel  and "fade in" on the new panel.

Thanks.

**Comment by wan54** [comment]
I wanted to use framer-motion's `AnimatePresence`. Had to do it this way but unsure if this is the "chakra" way because I removed `TabPanel` and use the regular stack.

`tabIndex` is updated via the `onChange` event.

```
<TabPanels>
      <AnimatePresence exitBeforeEnter initial={false}>
	      {tabs.map(
		      ({ title }, i) =>
			      i === tabIndex && (
				      <CrossFadeAnimation duration={1} key={title}>
					      <MyTabPanel
						      title={title}
					      />
				      </CrossFadeAnimation>
			      )
	      )}
      </AnimatePresence>
</TabPanels>
```

**Reply by with-heart** [reply]
This is going to cause problems with panel selection and accessibility, because `TabPanels` maps its `children` to pass a few props: https://github.com/chakra-ui/chakra-ui/blob/500b1eea0876909b670c103ffa354b2b77ff024f/packages/tabs/src/use-tabs.ts#L361-L370

You'd like need to make your own custom `TabPanels` component that uses `useTabPanels` but renders `children` wrapped in `AnimatePresence`.
